README.md

```diff
@@ -1,10 +1,80 @@
+ ***\*NOTE\****: This package is in technical preview, and breaking API changes are to be expected.
 
-`Characters` are strings viewed as sequences of *user-perceived character*s,
-also know as [Unicode (extended) grapheme clusters](https://unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries).
+[Characters][] are strings viewed as sequences of **user-perceived character**s, also know as [Unicode (extended) grapheme clusters][Grapheme Clusters].
+
+The `Characters` class allows access to the individual characters of a string, and a way to navigate back and forth between them using a `CharacterRange`.
+
+## Unicode Characters and Representations
+
+There is no such thing as plain text.
+
+Computers only know numbers, so any "text" on a computer is represented by numbers, which are again stored as bytes in memory.
+
+The meaning of those bytes are provided by layers of interpretation, building up to the *glyph*s that the computer displays on the screen.
+
+| Abstraction           | Dart Type                                                    | Usage                                                        | Example                                                      |
+| --------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| Bytes                 | [ByteBuffer][],<br />[Uint8List][]                           | Physical layout: Memory or network communication.            | `file.readAsBytesSync()`                                     |
+| [Code units][]        | [Uint8List][] ( UTF&#x2011;8)<br />[Uint16List][], [String][] (UTF&#x2011;16) | Standard formats for<br /> encoding code points in memory.<br />Stored in memory using one (UTF&#x2011;8) or more (UTF&#x2011;16) bytes. One or more code units encode a code point. | `string.codeUnits`<br />`string.codeUnitAt(index)`<br />`utf8.encode(string)` |
+| [Code points][]       | [Runes][]                                                    | The Unicode unit of meaning.                                 | `string.runes`                                               |
+| [Grapheme Clusters][] | [Characters][]                                               | Human perceived character. One or more code points.          | `string.characters`                                          |
+| [Glyphs][]            |                                                              | Visual rendering of grapheme clusters.                       | `print(string)`                                              |
+
+A Dart `String` is a sequence of UTF-16 code units, just like strings in JavaScript and Java. The runtime system decides on the underlying physical representation.
+
+That makes plain strings inadequate when needing to manipulate the text that a user is viewing, or entering, because they are working at the grapheme cluster level.
```

Comment:
Is "they" here the user? Then maybe "because humans perceive characters at the grapmeme cluster level"?

Reply:
"They" actually refers to "plain strings". Will rewrite to make unambiguous.